src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -120,6 +120,26 @@ def get(self, svc_id):
             'osd_metadata': mgr.get_metadata('osd', svc_id),
         }
 
+    @RESTController.Resource('GET')
+    @handle_send_command_error('osd')
+    def histogram(self, svc_id):
+        """
+        :return: Returns the requested data. The `histogram` key may contain a
+                 string with an error that occurred if the OSD is down.
+        """
+        try:
+            histogram = CephService.send_command(
+                'osd', srv_spec=svc_id, prefix='perf histogram dump')
+        except SendCommandError as e:  # pragma: no cover - the handling is too obvious
+            if 'osd down' in str(e):  # pragma: no cover - no complexity there
+                histogram = str(e)
```

Comment:
As a user i do not want to validate the returned data to ensure the data looks like the expected. If the endpoint returns an `dict` in case of success and a `str` in case of an error it is hard to process this data without checks. Is it really necessary to return a string in case the OSD is down? The error message yyou return here is part of the exception anyway. So the user can extract that information itself.

Reply:
Thanks @votdev .! Made the changes accordingly.